src/bob/states.clj

```diff
@@ -20,41 +20,21 @@
             [hikari-cp.core :as h]
             [clj-docker-client.core :as docker]
             [environ.core :as env]
-            [taoensso.timbre :as log]))
+            [taoensso.timbre :as log]
+            [bob.config :refer [config]]))
 
-(defn int-from-env
-  [key default]
-  (try
-    (Integer/parseInt (get env/env key (str default)))
-    (catch Exception _ default)))
-
-(defonce db-host (get env/env :bob-db-host "localhost"))
-(defonce db-port (int-from-env :bob-db-port 5432))
-(defonce db-user (get env/env :bob-db-user "bob"))
-(defonce db-name (get env/env :bob-db-name "bob"))
-
-(defonce docker-uri (get env/env :bob-docker-uri "unix:///var/run/docker.sock"))
-(defonce connect-timeout (int-from-env :bob-connect-timeout 1000))
-(defonce read-timeout (int-from-env :bob-read-timeout 30000))
-(defonce write-timeout (int-from-env :bob-write-timeout 30000))
-(defonce call-timeout (int-from-env :bob-call-timeout 40000))
-
-(defonce conn {:uri docker-uri
-               :timeouts {:connect-timeout connect-timeout
-                          :read-timeout    read-timeout
-                          :write-timeout   write-timeout
-                          :call-timeout    call-timeout}})
+(defonce conn (:docker config))
 
 (defonce images (docker/client {:category :images :conn conn}))
 (defonce containers (docker/client {:category :containers :conn conn}))
 (defonce commit (docker/client {:category :commit :conn conn}))
 
 (m/defstate data-source
   :start (let [data-source (h/make-datasource {:adapter            "postgresql"
-                                               :username           db-user
-                                               :database-name      db-name
-                                               :server-name        db-host
-                                               :port-number        db-port
+                                               :username           (get-in config [:postgres :user])
```

Comment:
these values can be defined as `(defonce ...)`? being used at two places.

Reply:
just wanted them not to clutter the namespace but will add again